### DISCUSSION TITLE
Seeking to contribute

### DISCUSSION THREAD

**Discussion by Digitl-Alchemyst** [tailwindlabs/tailwindcss#14155] [Ideas]
I am currently working with a set of custom plugins to extend the functionality of Tailwind CSS. While I am comfortable with modifying code, submitting pull requests, and other aspects of development, I find the Tailwind codebase quite unfamiliar and am unsure where to start.

Could anyone recommend resources or provide guidance on how to navigate the Tailwind codebase, specifically in understanding how CSS classes are set up and linked to their corresponding Tailwind definitions? Any assistance in pointing me to the relevant parts of the codebase would be greatly appreciated.

Thank you for your help.

**Comment by florence0239** [comment]
Hello, [GMSocrates](https://www.gm-socrates.com)

Navigating the Tailwind CSS codebase can indeed be a bit daunting at first, but there are some great resources and strategies to help you get started:

Tailwind CSS Documentation: The official documentation is a great place to start. It provides detailed explanations of the core concepts, utility-first fundamentals, and customization options. You can find it here.
Understanding the Codebase:
Core Concepts: Familiarize yourself with the core concepts of Tailwind CSS, such as utility-first design, responsive design, and dark mode. This will help you understand how classes are structured and used.
Configuration: Look into the tailwind.config.js file. This is where you can customize your Tailwind setup, including adding custom plugins.
Source Code: The source code for Tailwind CSS is available on GitHub. Browsing through the repository can give you insights into how different parts of the framework are implemented. You can find it here.
Community Resources:
Blog Posts and Tutorials: There are many blog posts and tutorials that walk through specific aspects of Tailwind CSS. For example, [this article](https://www.kindacode.com/article/tailwind-css-create-a-responsive-top-navigation-menu/) provides a practical example of creating a responsive navigation menu with Tailwind CSS
Practical Examples: Building small projects or components using Tailwind CSS can help you get a hands-on understanding of how everything fits together. The Tailwind CSS documentation includes many examples that you can use as a starting point.


Hope this will help you.
Best regards,
florence0239
